parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetFileWriter.java

```diff
@@ -766,4 +772,135 @@ public void testWriteMetadataFileWithRelativeOutputPath() throws IOException {
     ParquetFileWriter.writeMetadataFile(conf, relativeRoot, footers, JobSummaryLevel.ALL);
   }
 
+  @Test
+  public void testColumnIndexWriteRead() throws Exception {
+    File testFile = temp.newFile();
+    testFile.delete();
+
+    Path path = new Path(testFile.toURI());
+    Configuration configuration = new Configuration();
+
+    ParquetFileWriter w = new ParquetFileWriter(configuration, SCHEMA, path);
+    w.start();
+    w.startBlock(4);
+    w.startColumn(C1, 7, CODEC);
+    w.writeDataPage(7, 4, BytesInput.from(BYTES3), STATS1, BIT_PACKED, BIT_PACKED, PLAIN);
```

Comment:
I guess, you've checked my change by rebasing to the actual master where STATS1 and STATS2 are removed and exchanged to EMPTY_STATS. In the current change STATS1 and STATS2 still exist so it is correct as is. 
I did not rebase this change yet because it would cause loosing the pointers of the review findings. I'll do the rebase as soon as we'll have the required parquet-format release and this change can be finalized.

Reply:
Yes, I build the branch which merging your patch, and found the issue.

Reply:
Thanks for drawing my attention to this one. I'll keep in mind when I'm rebasing.